packages/api/src/services/context.servce.spec.ts

```diff
@@ -0,0 +1,116 @@
+import {ContextService} from './context.service';
+
+// Expose protected methods for testing
+class TestContextService extends ContextService {
+  public updateProperty<T>(propertyName: string, value: T): void {
+    this.updateContextProperty(propertyName, value);
+  }
+
+  public getProperty<T>(propertyName: string): T | undefined {
+    return this.getContextPropertyValue(propertyName);
+  }
+
+  public subscribeToProperty<T>(propertyName: string, callback: (value: T | undefined) => void): () => void {
+    return this.subscribe(propertyName, callback);
+  }
+}
+
+describe("ContextService", () => {
+  let contextService: TestContextService;
+
+  beforeEach(() => {
+    contextService = new TestContextService();
+  });
+
+  test('should be called immediately when the callback function is invoked with the last value of the property', () => {
+    const valueOfTestProperty = {a: 1, b: [1, 2]};
+    const propertyName = 'testProperty';
+
+    // Given the property is updated before the callback function is registered
+    contextService.updateProperty(propertyName, valueOfTestProperty);
+
+    // When a callback function is subscribed
+    const callBackFunction = jest.fn();
+    contextService.subscribeToProperty(propertyName, callBackFunction);
+
+    // Then the callback function should be called with the last value
+    expect(callBackFunction.mock.lastCall[0]).toEqual(valueOfTestProperty);
```

Comment:
`expect(callBackFunction).toHaveBeenCalledWith(valueOfTestProperty);`  or 
`expect(callBackFunction).toHaveBeenLastCalledWith(valueOfTestProperty);` depending on what you need, will be a bit more readable here

Reply:
Done